foundations/introduction/asking_for_help.md

```diff
@@ -12,19 +12,21 @@ This section contains a general overview of topics that you will learn in this l
 
 ### Tips for Getting the Best Help Possible
 
-#### <span id="provide-code-knowledge-check">Always Provide Your Code and the Surrounding Context</span>
+#### Always Provide Your Code and the Surrounding Context
 
-You can ask theoretical, vague questions, but you’re going to get theoretical, vague answers that probably won't be helpful to you and cause frustration on both ends as you and the person helping you dance around the real issue. The only time code doesn’t need to be provided is if you are asking a purely conceptual question outside of a specific project or code snippet. Additionally, provide as much context as is necessary, and zero in on the specific problem, such as pointing people to a specific function or line number in your code so people can give you the most relevant answers and quickly assist you.
+When asking a question it is essential to provide your code, error message, terminal command, server output, etc. You should provide as much context as possible and zero in on the specific problem, such as pointing people to a specific function or line number in your code. You should try to make it as easy as possible for others to help you.
+
+If your question does not provide any code or other relevant information, you will only receive theoretical and vague answers. This often causes frustration for everyone. You may become frustrated because the answers are not helpful. Those helping may become frustrated because as you add new information, the meaning of your original question changes and they wasted time trying to answer it. If you truly want to ask a purely conceptual question, you should indicate this in your question.
```

Comment:
```suggestion
If your question does not provide any code or other relevant information, this will often create unnecessary back and forth conversation to gather all the required information. If people are guessing what your problem is because they don't have all the details, their answers might be vague or technical, or not even the answer to your actual problem. This often causes frustration for everyone. You may become frustrated because the answers are not helpful. Those helping may become frustrated because as you add new information, the meaning of your original question changes and they wasted time trying to answer it. If you truly want to ask a purely conceptual question, you should indicate this in your question.
```

I reworded this a bit to add the "back and forth conversation". I often see this happening in our discord and they can sometimes make the path to an answer 20 messages long instead of the few it would have taken if the problem was clear from the start.

Reply:
I like the addition of the back and forth conversation, but the second sentence added a bit of repetition, since it also mentions that the answers are not helpful. Plus the back and forth conversation is the "this" that causes frustration. What do you think of this?

```suggestion
If your question does not provide any code or other relevant information, this will often create unnecessary back and forth conversation to gather all the required information. This often causes frustration for everyone. You may become frustrated because the answers are not helpful. Those helping may become frustrated because as you add new information, the meaning of your original question changes and they wasted time trying to answer it. If you truly want to ask a purely conceptual question, you should indicate this in your question.
```

Reply:
I see what you mean. I added the `back and forth` and `guessing` part based off on what I often see happening in our discord. But they do indeed translate to the same thing, both causing unnecessary back and forth. I think your rewrite captures that nicely.

Reply:
Hey @ManonLef. Did you see my version?

Reply:
I did, but I want to make sure Rachel can deal with these in succession and can maintain the birdseye view. Since it seems you reviewed the same text as I did. 

I do feel confident about this addition I made that is not reflected in your review. I added that because this document is the one we link to in our discord on a daily basis when guiding people with their issues. 

But to not overcomplicate this thread, I will leave the ownership with Rachel and she can review your suggestion as well.

Reply:
Ok gotcha 😁